### DISCUSSION TITLE
Idea: add custom local model(HTTP backend) for complex agents

### DISCUSSION THREAD

**Discussion by nfedyashev** [karthink/gptel#384] [Ideas]
depending on how hard it might be to add such feature custom(local) models might be a really powerful way of not just using LLMs but building own custom agentic workflows on top of it.

For gptel it could be just like another backend to send HTTP requests to.

Custom HTTP backend could have any kind of workflow - knowledge base injection, tool use, custom guard rails etc. As long as it responds in the expected format, it is fine.

Somewhat related issue #379 but in this case user is not limited to AWS Lambda as only action group option like AWS Bedrock limits it.

**Comment by karthink** [comment]
At the user-level that's one way to do it, yeah.  But this functionality should exist in gptel first, and I won't know if it fits in the backend framework until I implement a few of them.
